--- a/src/main/java/imgui/gl3/ImGuiImplGl3.java
+++ b/src/main/java/imgui/gl3/ImGuiImplGl3.java
@@ -194,6 +194,11 @@
 
             for (int cmdBufferIdx = 0; cmdBufferIdx < drawData.getCmdListCmdBufferSize(cmdListIdx); cmdBufferIdx++) {
                 drawData.getCmdListCmdBufferClipRect(clipRect, cmdListIdx, cmdBufferIdx);
+
+                final float clipRectX = (clipRect.x - displayPos.x) * framebufferScale.x;
+                final float clipRectY = (clipRect.y - displayPos.y) * framebufferScale.y;
+                final float clipRectZ = (clipRect.z - displayPos.x) * framebufferScale.x;
+                final float clipRectW = (clipRect.w - displayPos.y) * framebufferScale.y;
 
                 if (clipRectX < fbWidth && clipRectY < fbHeight && clipRectZ >= 0.0f && clipRectW >= 0.0f) {
                     // Apply scissor/clipping rectangle
